### PR TITLE
Update readme-1

### PR DESCRIPTION
**Pull request by gkGaneshR**
1. Why is this change necessary ?
fixes: openebs/openebs#1293

2. How does this change address the issue ?
The Readme file is updated.

Signed-off-by: gkGaneshR <gkganesh126@gmail.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,11 @@
-# cstor
+﻿cstor
+This repository hosts the kubernetization of cstor components.
+
+Cstor is the heart of OpenEBS that acts as interface between maya and underlying zfs filesystem. This repository creates images containing zpool and zfs binaries. While kubernetizing, it runs as pod comprising of Main container(controller for replica) and sidecar container(client for zfs and zpool).
```
Comment (mahebbar) [inline README.md]: acts as an...Maya and the underlying
runs as a pod
comprising of a
Main = main
container (...
container (client...

File: README.md [hunk 2]
```diff
@@ -1 +1,11 @@
...
+The means of communication between zfs component and maya component(k8s-volume – ref: https://github.com/openebs/maya)  is via Custom Resouce Definition(CRD).
```
Comment (mahebbar) [inline README.md]: component (k8s...
global change - space before opening brackets.

File: README.md [hunk 3]
```diff
@@ -1 +1,11 @@
...
+
+The repo includes testing where the cstor-sidecar watches for creation, deletion and updation of CstorCrd and performs zpool and zfs operations accordingly.
```
Comment (mahebbar) [inline README.md]: repo = repository
deletion, and...

File: README.md [hunk 4]
```diff
@@ -1 +1,11 @@
...
+
+![alt text](https://raw.githubusercontent.com/openebs/cstor/update-readme-1/images/sidecar-design.jpg)
+
+The design pattern followed here is Sidecar as well as Ambassador pattern where the zfs operations on one container is visible to the other through a shared volume.
```
Comment (mahebbar) [inline README.md]: is visible = are visible

**Comment by gkGaneshR** [comment]
@mahebbar Thanks for the review, Added commit

**Pull request merged**